app/client/src/components/shared/AddDataWidget.SearchPanel.tsx

```diff
@@ -740,59 +790,60 @@ function ResultCard({ result }: ResultCardProps) {
   /**
    * Adds non-tots layers as reference portal layers.
    */
-  function addRefLayer() {
+  const addRefLayer = useCallback(async () => {
     if (!mapView?.map) return;
 
     setStatus('loading');
 
-    Layer.fromPortalItem({
-      portalItem: new PortalItem({
-        id: result.id,
-      }),
-    }).then((layer) => {
-      // setup the watch event to see when the layer finishes loading
-      const newWatcher = watchUtils.watch(
-        layer,
-        'loadStatus',
-        (loadStatus: string) => {
-          // set the status based on the load status
-          if (loadStatus === 'loaded') {
-            setStatus('');
-
-            // set the min/max scale for tile layers
-            if (layer.type === 'tile') {
-              const tileLayer = layer;
-              tileLayer.minScale = 0;
-              tileLayer.maxScale = 0;
-            }
-
-            if (mapView) {
-              layer.visible = true;
-
-              // make all child layers visible, if applicable
-              if (layer.layers) {
-                layer.layers.items.forEach((tempLayer) => {
-                  tempLayer.visible = true;
-                });
-              }
-              if (layer.sublayers) {
-                layer.sublayers.items.forEach((tempLayer) => {
-                  tempLayer.visible = true;
-                });
-              }
-            }
-          } else if (loadStatus === 'failed') {
-            setStatus('error');
+    const layer: Layer | __esri.TileLayer | __esri.GroupLayer =
+      await Layer.fromPortalItem({
+        portalItem: new PortalItem({
+          id: result.id,
+        }),
+      });
```

Comment:
```suggestion
    Layer.fromPortalItem({
      portalItem: new PortalItem({
        id: result.id,
      }),
    }).then((layer) => {
```

Continuation of above.